Controller/Index/PaymentActionAbstract.php

```diff
@@ -0,0 +1,321 @@
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    public function execute()
+    {
+        if (!$this->configHelper->isActive()) {
+            $this->logger->error(
+                // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                basename(get_class($this)) . ' trigger error: Module is not active'
+            );
+            return $this->setResult(
+                __('Not Found: The required SwedbankPay Payments resources do not seem to be available'),
+                404
+            );
+        }
+
+        /** @noinspection PhpUndefinedMethodInspection */
+        $requestBody = json_decode($this->getRequest()->getContent());
+
+        $this->eventManager->dispatch(
+            $this->getFullEventName('before'),
+            ['requestBody' => $requestBody]
+        );
+
+        try {
+            $eventArgs = [];
+            foreach (array_keys($this->getEventArgs()) as $arg) {
+                $eventArgs[$arg] = $requestBody->{$arg};
+            }
+            $this->setEventArgs($eventArgs);
+            $this->logger->debug(
+                // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                basename(get_class($this)) . ' triggered',
+                ['eventArgs' => $eventArgs, 'requestBody' => (array)$requestBody]
+            );
+        } catch (\Exception $exception) {
+            $this->logger->error(
+                // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                basename(get_class($this)) . ' trigger error: Missing arguments in request body',
+                [
+                    'eventArgs' => (array) $this->getEventArgs(),
+                    'requestBody' => (array) $requestBody
+                ]
+            );
+            return $this->setResult(__('Bad Request: Missing argument(s)'), 400);
+        }
+
+        if (is_callable($this->getEventMethod())) {
+            $this->logger->debug(
+                'Calling event method: ' . $this->getEventMethodString(),
+                ['methodArgs' => $eventArgs]
+            );
+
+            $returnValue = null;
+            $errorMessage = '';
+
+            try {
+                // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                $result = $returnValue = call_user_func_array($this->getEventMethod(), $eventArgs);
+                if (is_scalar($result) && !is_string($result)) {
+                    $result = $result ? 'success' : 'error';
+                }
+            } catch (\Exception $exception) {
+                $result = 'error';
+                $errorMessage = $exception->getMessage();
+            }
+
+            if ($result == 'error') {
+                $this->logger->error(
+                    // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                    basename(get_class($this)) . ' trigger error: Failed to execute event method',
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function basename() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-payments/pullRequest?prid=5056328)